Components documentation improvements (#2121)
withastro/starlight: docs/src/content/docs/components/icons.mdx

---
title: Icons
description: Learn how to display icons in Starlight.
---

import { Icon } from '@astrojs/starlight/components';

> To display icons from a set of [common icons](/reference/icons/#all-icons), use the `<Icon>` component.

Reviewer: Couple of nits here:
- you can’t use “a set” — there’s only one supported set currently, so we should use a definite article (or like in this case specify “our” icons
- I think rather than “common”, which in English often means “widespread” or “frequent”, maybe something like “built-in” is a better way to describe these (mirrors the component language too “built-in components” / “built-in icons”)

```suggestion
To display icons from Starlight’s [built-in icon set](/reference/icons/#all-icons), use the `<Icon>` component.
```

import Preview from '~/components/component-preview.astro';

<Preview>

<Icon
	slot="preview"
	name="open-book"
	color="var(--sl-color-text-accent)"
	size="4rem"
/>

</Preview>

## Import

```tsx
import { Icon } from '@astrojs/starlight/components';
```

## Usage

Display an icon using the `<Icon>` component.

> An icon requires a [`name`](#name) set to [one of Starlight’s built-in icons](/reference/icons/#all-icons) and can optionally include a [`label`](#label) to provide context for screen readers.

Reviewer: Ha, you got it perfect here already! 😄

The [`size`](#size) and [`color`](#color) attributes can be used to adjust the icon’s appearance using CSS units and color values.
The [`class`](#class) attribute can be used to add custom CSS classes to the icon.

<Preview>

```mdx
# src/content/docs/example.mdx

import { Card } from '@astrojs/starlight/components';

<Icon name="star" color="goldenrod" size="2rem" />
<Icon name="rocket" color="var(--sl-color-text-accent)" />
```

<Fragment slot="markdoc">

```markdoc
# src/content/docs/example.mdoc

{% icon name="star" color="goldenrod" size="2rem" /%}
{% icon name="rocket" color="var(--sl-color-text-accent)" /%}
```

</Fragment>

<Fragment slot="preview">
	<Icon name="star" color="goldenrod" size="2rem" />
	<Icon name="rocket" color="var(--sl-color-text-accent)" />
</Fragment>

</Preview>

## `<Icon>` Props

**Implementation:** [`Icon.astro`](https://github.com/withastro/starlight/blob/main/packages/starlight/user-components/Icon.astro)

The `<Icon>` component accepts the following props:

### `name`

**required**  
**type:** `string`

The name of the icon to display set to [one of Starlight’s built-in icons](/reference/icons/#all-icons).

### `label`

**type:** `string`

> An optional label to provide context for screen readers.

Reviewer: Might be worth documenting the full accessibility behaviour here while we’re at it:

```suggestion
An optional label to provide context for assistive technologies, such as screen readers.

When `label` is not set, the icon will be completely hidden from assistive technologies.
In this case, make sure the context is still understandable without the icon.
For example, a link containing only the icon **must** include the `label` attribute in order to be accessible, but if a link contains text and the icon is purely decorative, omitting the `label` may make sense.
```

### `size`

**type:** `string`

The size of the icon using CSS units.

### `color`

**type:** `string`

The color of the icon using a CSS color value.

### `class`

**type:** `string`

> Defines custom CSS classes to add to the icon.

Reviewer: Micro-nit, but the other entries just say “the size…”, “the color…”, so we can probably skip the “defines” here too:

```suggestion
Custom CSS classes to add to the icon.
```
